charts/nucliadb_writer/templates/writer.vs.yaml

```diff
@@ -10,22 +10,68 @@ spec:
   http:
     - name: nucliadb_writer
       match:
+        # KBs (only for onprem)
         - uri:
             regex: '^/api/v\d+/kb/[^/]+'
           method:
             regex: "POST|PATCH|PUT|DELETE|OPTIONS"
-        - uri:
-            regex: '^/api/v\d+/kb/[^/]+/(resource|slug|resources|entitiesgroup|labelset|upload|custom-synonyms|export|import).*'
-          method:
-            regex: "POST|PATCH|PUT|DELETE|OPTIONS"
+
+        # KB config
         - uri:
             regex: '^/api/v\d+/kb/[^/]+/configuration'
           method:
             regex: "POST|PATCH|OPTIONS"
         - uri:
-            regex: '^/api/v\d+/kb/[^/]+/vectorsets/.*'
+            regex: '^/api/v\d+/kb/[^/]+/vectorsets/[^/]+$'
           method:
             regex: "POST|DELETE|OPTIONS"
+
+        # resources
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/resources'
+          method:
+            regex: "POST|OPTIONS"
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+$'
+          method:
+            regex: "PATCH|DELETE|OPTIONS"
+
+        # fields
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+/(text|file|link|conversation)/[^/]+$'
+          method:
+            regex: "PUT|DELETE|OPTIONS"
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+/conversation/[^/]+/messages'
+          method:
+            regex: "PUT|OPTIONS"
+
+        # upload
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/upload'
+          method:
+            regex: "POST|OPTIONS"
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+/file/[^/]+/upload'
+          method:
+            regex: "PUT|DELETE|OPTIONS"
+
+        # reprocess/reindex
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+/(reprocess|reindex)'
+          method:
+            regex: "POST|OPTIONS"
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(resource|slug)/[^/]+/file/[^/]+/reprocess'
+          method:
+            regex: "POST|OPTIONS"
+
+        # misc
+        - uri:
+            regex: '^/api/v\d+/kb/[^/]+/(entitiesgroup|labelset|custom-synonyms|export|import).*'
+          method:
+            regex: "POST|PATCH|PUT|DELETE|OPTIONS"
+
```

Comment:
Would also be good to remove `.*` at the end of the tusupload routes. But that's way less likely to collide, so not urgent.